services/ml/compilation_delegate_mps.mm

```diff
@@ -50,6 +52,17 @@
     dimensions[3] = extend_channels;
   }
 
+  // The MPSImage will exceed the maximum allowed number of 2048 slices that
+  // need to be reshaped and reshape to expected dimensions when using. So the
+  // operation of using the input data must be Reshape Operation.
+  if (dimensions[3] > MAX_TEXTURE_ARRAY &&
+      dimensions[3] == product(dimensions)) {
```

Comment:
Can remove this check? and make this work for h or w is not 1.

Reply:
Thanks, done